pkg/inject/config_override.go

```diff
@@ -0,0 +1,345 @@
+package inject
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	l5dcharts "github.com/linkerd/linkerd2/pkg/charts/linkerd2"
+	"github.com/linkerd/linkerd2/pkg/k8s"
+	"github.com/linkerd/linkerd2/pkg/util"
+	log "github.com/sirupsen/logrus"
+	k8sResource "k8s.io/apimachinery/pkg/api/resource"
+)
+
+type ConfigLookup interface {
+	GetConfigOverride(string) (string, bool)
+	GetInboundPorts() string
```

Comment:
This is only used in `GetOverriddenValues` to modify the resulting values.  This could be taken out of `GetOverridenValues` entirely and instead the caller can do this modification afterwards.

Reply:
Yeah, small downside that we have to remove a small test case from a unit test for `GetOverridenValues`, but luckily we still test inbound ports when we generate the patch so we should be good.